app/jobs/onceoff/recover_post_uploads.rb

```diff
@@ -0,0 +1,10 @@
+require_dependency "upload_recovery"
+
+module Jobs
+  class RecoverPostUploads < Jobs::Onceoff
+    def execute_onceoff(args)
+      grace_period = SiteSetting.purge_deleted_uploads_grace_period_days.days.ago
+      UploadRecovery.new.recover(Post.where("baked_at >= ?", grace_period))
```

Comment:
hmm so you pass in a relation... I hope the job is not materializing all those posts cause memory will explode, what happens if a single post explodes doe the whole onceoff fail?

Reply:
Yup if a single post explodes, the onceoff fail and we can figure out why. I'm doing a `find_each` so no worries about it materializing everything.